payjoin-directory/src/lib.rs

```diff
@@ -426,6 +427,16 @@ fn not_found() -> Response<BoxBody<Bytes, hyper::Error>> {
     res
 }
 
+async fn handle_ohttp_gateway_get(
+    ohttp: &Arc<Mutex<ohttp::Server>>,
+    query: &str,
+) -> Result<Response<BoxBody<Bytes, hyper::Error>>, HandlerError> {
+    match query {
+        "allowed-purposes" => get_ohttp_allowed_purposes().await,
```

Comment:
```suggestion
        "allowed_purposes" => get_ohttp_allowed_purposes().await,
```

If we're using an underscore as discussed [here in ohttp-relay #58](https://github.com/payjoin/ohttp-relay/pull/58#discussion_r1988359678) this needs to be an underscore.